app/scripts/controllers/app-state-controller.ts

```diff
@@ -300,26 +466,26 @@ export class AppStateController extends EventEmitter {
    * @param shouldShowUnlockRequest - Whether the extension notification
    * popup should be opened.
    */
-  waitForUnlock(resolve: () => void, shouldShowUnlockRequest: boolean): void {
+  #waitForUnlock(resolve: () => void, shouldShowUnlockRequest: boolean): void {
     this.waitingForUnlock.push({ resolve });
-    this.emit(METAMASK_CONTROLLER_EVENTS.UPDATE_BADGE);
+    this.messagingSystem.publish('AppStateController:unlockChange');
```

Comment:
Good decision on renaming this event, this is more descriptive 👍 Confusing to reuse the update badge event name here.